javascript/javascript2/week1/currencyConverter.js

```diff
@@ -0,0 +1,82 @@
+// currency-converter-script.js
+
+document.addEventListener("DOMContentLoaded", () => {
+  const exRates = {
+    timestamp: Date.now(),
+    base: "USD",
+    date: new Date().toISOString().split("T")[0],
+    exRates: {
+      USD: 1.23396,
+      GBP: 0.882047,
+      EUR: 0.923987,
+      CAD: 1.36325,
+    },
+  };
+
+  const addNewRateform = document.getElementById("addNewRateform");
+  const convertCurrencyForm = document.getElementById("convertCurrencyForm");
+  const updateRateForm = document.getElementById("updateRateForm");
+  const addRateResult = document.getElementById("addRateResult");
+  const convertCurrencyResult = document.getElementById(
+    "convertCurrencyResult"
+  );
+  const updateRateResult = document.getElementById("updateRateResult");
+
+  addNewRateform.addEventListener("submit", (event) => {
+    event.preventDefault();
+    const baseCurrency = document
+      .getElementById("base-currency")
+      .value.toUpperCase();
+    const targetCurrency = document
+      .getElementById("target-currency")
+      .value.toUpperCase();
+    const rate = parseFloat(document.getElementById("exchange-rate").value);
+
+    exRates.base = baseCurrency;
+    exRates.exRates[targetCurrency] = rate;
+
+    console.log("Inserted new rate:", exRates);
+    addRateResult.textContent = `New rate is: 1 ${baseCurrency} = ${rate} ${targetCurrency}`;
+  });
+
+  convertCurrencyForm.addEventListener("submit", (event) => {
```

Comment:
You can extract these callbacks to make your code more organised:
```suggestion
  function onConversionFormSubmitted(event) {
    // ...
  }
  convertCurrencyForm.addEventListener("submit", onConversionFormSubmitted)
```

Reply:
Suggestions implemented. Thank you!